mavros_extras/src/plugins/esc_telemetry.cpp

```diff
@@ -21,6 +25,90 @@ namespace mavros
 {
 namespace extra_plugins
 {
+// for all the dignostics parameters
+struct Limits {
+	bool diag_enabled;
+	// Temperature checks
+	float temp_min_nok;
+	float temp_min_ok;
+	float temp_max_nok;
+	float temp_max_ok;
+	// Voltage checks
+	float volt_min_nok;
+	float volt_min_ok;
+	float volt_max_nok;
+	float volt_max_ok;
+	// Current checks
+	float curr_min_nok;
+	float curr_min_ok;
+	float curr_max_nok;
+	float curr_max_ok;
+	// RPM checks
+	int rpm_min_nok;
+	int rpm_min_ok;
+	int rpm_max_nok;
+	int rpm_max_ok;
```

Comment:
We can generate most of that fields by cog also. Will need to move fields list and extend it to have type.